kolena/workflow/_datatypes.py

```diff
@@ -146,6 +162,11 @@ def deserialize_field(
             )
 
         items = {f.name: deserialize_field(f, obj_dict.get(f.name, None)) for f in dataclasses.fields(cls) if f.init}
+        field_names = {f.name for f in dataclasses.fields(cls)}
+        if allow_extra(cls):
+            for key, val in obj_dict.items():
+                if key not in field_names:
+                    items[key] = val
```

Comment:
This may need more attention, as structured values are not correctly deserialized:

```py
from kolena.workflow import GroundTruth
from kolena.workflow.annotation import BoundingBox

gt = GroundTruth(bboxes=[BoundingBox(top_left=(1, 1), bottom_right=(2, 2))])

# succeeds
assert isinstance(gt.bboxes[0], BoundingBox)

# fails, bboxes is still List[Dict[str, ...]] instead of List[BoundingBox]
assert isinstance(GroundTruth._from_dict(gt._to_dict()).bboxes[0], BoundingBox)
```

Reply:
At this point, we are trying to instantiate an instance, based on class info, from a dictionary. For extra fields. We don't have type information, i.e. from `dataclass.fields`, as such can only keep what it is from the dict.

Let me know if I missed something.

Reply:
For typed data objects, we have `data_type`, which should be sufficient for our deserialization needs. Will need some extra handling here but for any non-scalar object, we should attempt to deserialize its members if any `data_type` is specified.